analytics/backend_sqs.go

```diff
@@ -0,0 +1,66 @@
+package analytics
+
+import (
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/ONSdigital/go-ns/log"
+	"github.com/aws/aws-sdk-go-v2/aws/external"
+	"github.com/aws/aws-sdk-go-v2/service/sqs"
+	"github.com/aws/aws-sdk-go-v2/service/sqs/sqsiface"
+)
+
+var _ ServiceBackend = &sqsBackend{}
+
+type sqsBackend struct {
+	sqsService  sqsiface.SQSAPI // *sqs.SQS
+	queueURL    string
+	sendMessage func(sqs.SendMessageRequest) (*sqs.SendMessageOutput, error)
+}
+
+// NewSQSBackend creates a new SQS backend for storing analytics data
+func NewSQSBackend(queueURL string) (ServiceBackend, error) {
+	cfg, err := external.LoadDefaultAWSConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	return &sqsBackend{sqs.New(cfg), queueURL, func(s sqs.SendMessageRequest) (*sqs.SendMessageOutput, error) {
+		return s.Send()
+	}}, nil
+}
+
+func (b *sqsBackend) Store(req *http.Request, url, term, listType, gaID string, gID string, pageIndex, linkIndex, pageSize float64) {
```

Comment:
Can these be put into a struct? Passing this many args around is nasty - makes the method ugly.

Reply:
good point - I'll try it but profile it before we change the PR, that could lead to lots of object creation and garbage collection since they'd all (probably) end up on the heap